### PR TITLE
Ma/create user conflict

### PR DESCRIPTION
**Pull request by markusahlstrand**
Added a new method to the users model for creating new users that throws a conflict error if the user already exists.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,44 +43,44 @@
...
+    "miniflare": "^3.20230821.0",
     "mjml": "^4.14.1",
-    "prettier": "^3.0.1",
-    "semantic-release": "^21.0.7",
+    "prettier": "^3.0.3",
+    "semantic-release": "^21.1.1",
     "ts-jest": "29.1.1",
     "ts-node": "^10.9.1",
-    "typescript": "^5.1.6",
+    "typescript": "^5.2.2",
```
Comment (danielgent) [inline package.json:83]: One thing I was thinking about with this is we don't actually have any idea if these upgrades will work on cloudflare right?

As our testing is in Jest on Node, and even then does it actually use all these libraries? :smile:   e.g. `@planetscale/database` and `kysely` are definitely not being tested *BUT* I have no idea how we'd test these on Cloudflare on a PR

*only* after merging we could run the login2 e2e tests.  If they pass then that's much better :shrug:

Comment (danielgent) [reply package.json:83]: Maybe we change the login2 tests cron job to run hourly :joy:

File: src/helpers/sql.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const parsedRange = parseRange(rangeRequest);
+
+  const data = await query
+    .selectAll()
+    .offset(parsedRange.from)
+    .limit(parsedRange.limit)
+    .execute();
+
+  let range;
+  if (parsedRange.entity) {
```
Comment (danielgent) [inline src/helpers/sql.ts:18]: I hate mutable variables and extra logic :smile: 


I :heart:  short circuiting

`if (!parsedRange.entity) return { data }`

File: src/routes/tsoa/applications.ts [hunk 3]
```diff
@@ -31,29 +31,22 @@ export class ApplicationsController extends Controller {
   public async listApplications(
     @Request() request: RequestWithContext,
     @Path("tenantId") tenantId: string,
-    @Header("range") range?: string,
+    @Header("range") rangeRequest?: string,
```
Comment (danielgent) [inline src/routes/tsoa/applications.ts:34]: ahhh, I think this PR should be pointing to `ma/users-search`? :thinking: 

I'll review the others first

File: package.json (hunk 1 above)
Comment (markusahlstrand) [reply package.json:83]: Yeah.. I thought I had an issue with Kysely that I thought was related to the version and did an update. Turned out not to be the case :)

File: src/models/User.ts [hunk 4]
```diff
@@ -250,6 +275,24 @@ export const userRouter = router({
...
+
+      const profile = await updateProfile(ctx, input);
+
+      await writeLog(ctx.state.storage, {
+        category: "update",
+        message: "User created",
+      });
+
+      return profile;
+    }),
```
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply src/models/User.ts:295]: Yes, we could add that :)

File: src/helpers/sql.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/helpers/sql.ts:18]: Yes, that would be better :)

**Pull request merged**